src/vuestic-theme/vuestic-components/va-rating/VaRating.vue

```diff
@@ -0,0 +1,301 @@
+      type: String,
+      default: 'fa fa-star'
+    },
+    halfIcon: {
+      type: String
+    },
+    emptyIcon: {
+      type: String
+    },
+    value: {
+      type: Number,
+      default: 1
+    },
+    readonly: {
+      type: Boolean
+    },
+    disabled: {
+      type: Boolean
+    },
+    numbers: {
+      type: Boolean
+    },
+    max: {
+      type: Number,
+      default: 5
+    },
+    size: {
+      type: String,
+      default: 'medium'
+    },
+    color: {
+      type: String,
+      default: 'success'
+    }
+  },
+  data () {
+    return {
+      lastHoverItemNumber: this.value,
+      isHovered: false,
+      tabindex: 0
+    }
+  },
+  computed: {
+    valueProxy: {
+      set (valueProxy) {
+        this.$emit('input', valueProxy)
+      },
+      get () {
+        return this.value
+      }
+    },
+    emptyIconComputed () {
+      return this.emptyIcon || this.icon + ' ' + 'va-rating__icons--empty'
+    },
+    halfIconComputed () {
+      return this.disabled || this.readonly ? '' : this.halfIcon
+    },
+
+    computedClasses () {
+      return {
+        'va-rating--success': this.color === 'success',
+        'va-rating--info': this.color === 'info',
+        'va-rating--danger': this.color === 'danger',
+        'va-rating--warning': this.color === 'warning',
+        'va-rating--gray': this.color === 'gray',
+        'va-rating--dark': this.color === 'dark',
+        'va-rating--disabled': this.disabled,
+        'va-rating--readonly': this.readonly,
+        'va-rating--medium': this.size === 'medium',
+        'va-rating--small': this.size === 'small',
+        'va-rating--large': this.size === 'large',
+      }
+    },
+  },
+  methods: {
+    getTabindex (value) {
+      if (!this.disabled) {
+        return value !== this.tabindex ? 0 : null
+      }
+    },
+    getItemsFontSize () {
+      if (!isNaN(this.size.substring(0, 1))) {
+        return this.size
+      }
+      if (this.size === 'medium') {
+        return 1 + 'rem'
+      }
+      if (this.size === 'large') {
+        return 1.5 + 'rem'
+      }
+      if (this.size === 'small') {
+        return 0.75 + 'rem'
+      }
+    },
+    onHover (itemNumber) {
+      if (this.halfIcon) {
+        this.lastHoverItemNumber = itemNumber
+      }
+    },
+    getIconClasses (itemNumber) {
+      const iconClass = this.emptyIcon || this.icon + ' ' + 'va-rating__icons--empty'
+      return this.compareWithValue(itemNumber) ? this.icon : iconClass
+    },
+    getIconSize () {
+      let k = 7 / 6 // coefficient, which controls connection between size and font-size
+      if (this.numbers) {
+        k = 4.5 / 7
+      }
+      const regEx = /[a-z]/gim
+      const size = this.getItemsFontSize()
+      const unit = size.match(regEx).join('')
+      return size.replace(regEx, '') * k + unit
+    },
+    setCurrentValue (item, value) {
+      if (this.readonly || this.disabled) {
+        return
+      }
+      if (item - this.valueProxy === 0.5) {
```

Comment:
valueProxy is for template usage convenience mostly. You can use value directly here.